frontends/p4/strengthReduction.cpp

```diff
@@ -198,6 +198,8 @@ const IR::Node *DoStrengthReduction::postorder(IR::Sub *expr) {
 const IR::Node *DoStrengthReduction::postorder(IR::Add *expr) {
     if (isZero(expr->right)) return expr->left;
     if (isZero(expr->left)) return expr->right;
+    if (expr->left->is<IR::Constant>())
```

Comment:
What about other commutative operators? Maybe we should have a reassociation pass before constant folding for all such ops, if there isn't one already.

Reply:
I think I'm having some extra patterns somewhere downstream... Let me find them.

Reply:
> What about other commutative operators? Maybe we should have a reassociation/canonicalisation pass before constant folding for all such ops, if there isn't one already.

There is a `Reassociation` pass that runs after this -- adding this swap for commutative ops makes it work in more cases.